### DISCUSSION TITLE
Snapshot delay

### DISCUSSION THREAD

**Discussion by d3vl1m3** [oblador/loki#384] [Q&A] [answered]
For specific stories, is it possible to add a delay to allow a story/animation to fully load in before taking the snapshot? 

I know there is the `isLokiRunning()` but it would be good as a quick fix for my time for now to be able to delay the snapshot being taken.

**Comment by techeverri** [comment]
You can use `@loki/create-async-callback` as explained here: https://github.com/oblador/loki/pull/258

**Reply by d3vl1m3** [reply]
Ah! Wait. Nope. I've looked at this a few times and still don't get it. If adding this to a mature library that example makes it look like I need to extend the component to work with this. 

Am I misunderstanding it or is this something that can be added in via config?

I don't want to add a callback, I want to set a global delay.

**Reply by d3vl1m3** [reply]
Nope. Can't get this to work even following what I thought was working in my answer. 

Do you have any examples of this working that don't use 'ofStories'?

**Reply by d3vl1m3** [reply]
Wait! Yes it is! What is happening!!?!? FML. I'm going to comb through it properly to see why I thought it wasn't working.

**Comment by d3vl1m3** [comment, marked as answer]
Okay. Figured it out without using the `storiesOf`. 

You can use this as a custom hook or a component - both seem to work well:

```jsx
// use-delay-capture.hook.jsx
import {useEffect} from 'react';
import isLokiRunning from '@loki/is-loki-running';
import createAsyncCallback from "@loki/create-async-callback";

const useDelayCapture = (delayInMs = 1000) => {
  useEffect(() => {
    if ( isLokiRunning() ) {
      const onDone = createAsyncCallback();
      const timer = setTimeout(() => onDone(), delayInMs);
      return () => clearTimeout(timer);
    }
  }, [delayInMs]);
}
export default useDelayCapture;
```

**OR**

```jsx
// delay-capture.component.jsx
import React, {useEffect} from 'react';
import isLokiRunning from '@loki/is-loki-running';
import createAsyncCallback from "@loki/create-async-callback";

const DelayCapture = (props) => {
  const { delayInMs = 1000} = props; 

  useEffect(() => {
    if ( isLokiRunning() ) {
      const onDone = createAsyncCallback();
      const timer = setTimeout(() => onDone(), delayInMs);
      return () => clearTimeout(timer);
    }
  }, [delayInMs]);

  return null;
}

export default DelayCapture;
```
---

Used accordingly in the story template function as:

```jsx
// example.stories.jsx
const Template = () => {
  useDelayCapture();
  // Or with a prop. e.g:
  // useDelayCapture(2000);

  return <VideoComponent />
}
```

OR 

```jsx
// example.stories.jsx
const Template = () =>  <>
  {/* Your component here first */}
  <FooBar/>

  {/* Then the delayed one holds off the screen capture whilst yours does its thing */}
  <DelayCapture />
  {/* Or with a prop. e.g: */}  
  {/* <DelayCapture delay={2000} /> */}
</> ;
```

Loki update now passes and the rendered capture shows the following:
![Screenshot 2022-07-29 at 11 37 19](https://user-images.githubusercontent.com/63494392/181742069-3c0288ff-7c3d-4bea-b81a-7b0034d5301a.png)
